core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java

```diff
@@ -216,6 +216,7 @@ public long takeSnapshot() {
         SAMPLING_LOG.warn("Failed to get raft group info: {}", e.getMessage());
       }
       long index = mSnapshotManager.maybeCopySnapshotFromFollower();
+      mSnapshotLastIndex = index;
```

Comment:
`index` maybe `-1`  while `DownloadState` is not one of `DOWNLOADED` `REQUEST_DATA` and `IDLE`

Reply:
```suggestion
      if (index != RaftLog.INVALID_LOG_INDEX) {
        mSnapshotLastIndex = index;
      }
```

Reply:
Good after committing this change.